primitives/src/v1/mod.rs

```diff
@@ -1054,6 +1055,12 @@ sp_api::decl_runtime_apis! {
 
 		/// Scrape dispute relevant from on-chain, backing votes and resolved disputes.
 		fn on_chain_votes() -> Option<ScrapedOnChainVotes<H>>;
+
+		/// Submits a PVF pre-checking statement into the transaction pool.
+		fn submit_pvf_check_statement(stmt: PvfCheckStatement, signature: ValidatorSignature);
```

Comment:
Other kind of statements we imported via inherents. How come we are using the transaction pool here?

Reply:
It was suggested by Rob [here](https://github.com/paritytech/polkadot/issues/3211). We never really discussed exactly this part, mostly because I agree with this. On one hand, we can afford routing those through the chain: we do not expect a lot of those and they are not too heavy. On the other hand, it greatly simplifies the whole construction. The next PR is for subsystem and you'll see how simple it is.

That being said, in future we can reconsider and implement it via a network protocol if such need arises.

Reply:
Not sure, I fully understand this - maybe a quick call tomorrow?

Reply:
sure, thing.